src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -190,17 +192,78 @@ public static int EncodeInPlace(Span<byte> buffer, int sourceLength)
             return encodedLength;
         }
 
+        #endregion
+
+        #region Decode
+
+        public static int ComputeDecodedLength(ReadOnlySpan<byte> source)
+        {
+            ref byte srcBytes = ref source.DangerousGetPinnableReference();
```

Comment:
Do you really need to get the ref? What's the per difference if you use the indexer? i.e. I think it's fine for us to use dangerous APIs when we operate on long sequences of bytes, but here we should be able to use the indexer. If we cannot, then maybe the JIT team can look into this.